src/main.rs

```diff
@@ -620,16 +628,17 @@ async fn inner(window: Window, gfx: Graphics, mut ev: EventStream) -> Result<(),
         .with(Position(Vector::new(600.0, 300.0)))
         .build();
 
+
+    // Adjust the viewport before first frame
+    adjust_viewport_to_window_size(gfx.borrow(), &mut world, window.borrow());
```

Comment:
Why call `window.borrow()` instead of just `&window`? This seems… weird.

Reply:
Definitely good point, thank you :)